### ISSUE TITLE
Documentation clarification about property failsafe values

### ISSUE THREAD

**Issue by fmunteanu** [acidanthera/bugtracker#1477]
This is related directly to documentation. if I understand correctly, there are no `default` values within OpenCore. Only `failsafe`, which tries to reduce the harm on your system when the configuration is corrupted.

`ocvalidate` is a great tool allowing us to determine if `config.plist` has any errors. For example, setting only `BIOSVersion` value, will make `ocvalidate` complain about the missing properties:

```xml
<key>PlatformInfo</key>
<dict>
    <key>SMBIOS</key>
    <dict>
        <key>BIOSVersion</key>
        <string>9999.0.0.0.0</string>
    </dict>
    <key>UpdateDataHub</key>
    <false/>
    <key>UpdateNVRAM</key>
    <false/>
    <key>UpdateSMBIOS</key>
    <true/>
    <key>UpdateSMBIOSMode</key>
    <string>Create</string>
    <key>UseRawUuidEncoding</key>
    <false/>
</dict>
```

Is it possible to document what is the actual OC behaviour when missing properties occur? I'm also interested to know if the missing properties are automatically populated from the actual OEM machine (`UpdateSMBIOS=True`). This will help understanding better the failsafe behaviour. @vit9696 if you could briefly explain the behaviour here, it will be much appreciated.

**Comment by vit9696** [comment]
Hi @fmunteanu,

Basically the situation with Configuration.pdf and OpenCore is similar to how a specification and an implementation work. The specification imposes some requirements, and the implementation must follow. There always are some grey areas in the specification, where it tries to be less verbose and allows flexibility for the implementation. In such cases the implementation may do what it sees fit, and this behaviour cannot be found in the specification.

At best the specification may try to give recommendations for the implementation, and these failsafe values are pretty much that. I.e. when the implementation does not know what to do, and it ends up in a situation that is not covered by the specification (i.e. essentially the requirements are no longer defined), it should still do something. For performance reasons it may not be practical to perform error checking of many things, in fact, some things cannot be checked at all depending on the time. It is also not so useful to just die silently from the user perspective.

Long story short, OpenCore tries to treat missing values as if they are set to failsafe values by the user. You should not rely on this, but this is what currently happens. In fact, this is what we try to do in general: treat missing fields as if they are set to what is specified in failsafe, but warning the user first of course. For SMBIOS failsafes are chosen in a way that the original OEM values will be preserved, most of the time.

Hope it helps.

**Comment by fmunteanu** [comment]
Thank you @vit9696, if I understand correctly, it is strongly recommended to define into `config.plist` all properties with their default failsafe values, in order to to avoid OC guessing/interpreting if a failsafe value or OEM system value should be used?

**Comment by vit9696** [comment]
Mostly correct. But I should clarify that "if a failsafe value or OEM system" is not right, OC will only guess about the user-entered value. Once it parses the plist it always has some value for all the properties (be that user-entered or failsafe). Some values for e.g. SMBIOS mean that OEM values are to be used. I.e. these are unrelated things.

**Comment by startergo** [comment]
Safely to assume then on real Macs those values are correct.
